articles/quickstart/webapp/php/01-login.md

```diff
@@ -11,159 +12,214 @@ topics:
-// index.php
-
-$laravelStore = new MyLaravelStore();
-$auth0 = new Auth0(array(
-    // ...
-    'store' => $laravelStore,
-    // ...
-));
+  if ($session === null) {
+    // The user isn't logged in.
+    echo 'Please <a href="/login">log in</a>.</p>';
+    return;
+  }
+
+  // The user is logged in.
+  print_r($session->user);
+
+  echo 'You can now <a href="/logout">log out</a>.</p>';
+}
+```
+
+We can access all the properties of our user's profile from the `user` property response, which is an array. So, for example, we can pull the user's nickname using `$session->user['nickname']`, or their email address from `$session->user['email']`. The structure is a [normalized user profile](https://auth0.com/docs/users/user-profile-structure), which you can learn more about [here](https://auth0.com/docs/users/normalized-user-profiles).
+
+It's important to note that the content of the user profile will vary depending on the social provider(s) you use, so you should never assume that a particular value will always be there within your application logic. Use PHP language constructs like `isset` or null coalescence to gracefully handle a value's presence, or lack thereof, for example:
+
+```PHP
+// ✋ We don't need to include this in our sample application, it's just an example.
+$name = $session->user['name'] ?? $session->user['nickname'] ?? $session->user['email'] ?? 'Unknown';
+```
+
+### Logging in
+
+Now let's create our /login route, which will use the Auth0 PHP SDK's `login()` method to setup the user session and return a customized URL to Auth0's Universal Login Page for this user to login.
+
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
+
+function onLoginRoute() {
+    // It's a good idea to reset user sessions each time they go to login to avoid "invalid state" errors, should they hit network issues or other problems that interrupt a previous login process:
+    $auth0->clear();
+
+    // Setup the user's session and generate a ULP URL:
+    $loginUrl = $auth0->login(ROUTE_URL_CALLBACK);
+
+    // Finally, redirect the user to the Auth0 Universal Login Page.
+    header("Location: " . $loginUrl);
+    exit;
+}
+```
+
+### Handling authentication callback
+
+After our users return from authenticating with the Auth0's Universal Login Page, they'll return to our sample application at our callback route, `/callback` which we'll handle in this step.
+
+When Auth0 passes our users back to us, it includes a few essential parameters in the query of the HTTP request. The Auth0 PHP SDK's `exchange()` method handles working with those, so finishing our authentication flow is straightforward:
+
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
+
+function onCallbackRoute() {
+    // Have the SDK complete the authentication flow:
+    $auth0->exchange(ROUTE_URL_CALLBACK);
+
+    // Finally, redirect our end user back to the / index route, to display their user profile:
```

Comment:
Should this be /index or are you separating it on purpose? The paragraph in the section below has the same typo at the end.
```suggestion
    // Finally, redirect our end-user back to the /index route, to display their user profile:
```

Reply:
I've updated the phrasing there a little to clarify